app/services/hyrax/workflow/permission_query.rb

```diff
@@ -257,8 +257,11 @@ def scope_users_for_entity_and_roles(entity:, roles:)
             .and(entity_responsibilities[:entity_id].eq(entity.id))
         )
 
-        # PostgreSQL requires an explicit cast from string to integer
-        cast = Arel::Nodes::NamedFunction.new "CAST", [agent_table[:proxy_for_id].as("integer")]
+        # Default to "integer" for adapters like Postgres and Sqlite, but cast
+        # to "signed" for Mysql. The type CAST causes a SQL syntax error for an
+        # unsupported type depending on which adapter is being used.
+        type = ActiveRecord::Base.connection.adapter_name.casecmp("mysql2").zero? ? "signed" : "integer"
```

Comment:
How about giving this a try: `ActiveRecord::Base.connection.type_to_sql(:integer)` that returns `"integer"` in postgres.  I suspect it'll return `unsigned` in mysql.